src/Compiler/FSComp.txt

```diff
@@ -1396,10 +1396,11 @@ tcAnonRecdInvalid,"Invalid Anonymous Record type declaration."
 tcAnonRecdCcuMismatch,"Two anonymous record types are from different assemblies '%s' and '%s'"
 tcAnonRecdFieldNameMismatch,"This anonymous record does not exactly match the expected shape. Add the missing fields %s and remove the extra fields %s."
 tcAnonRecdSingleFieldNameSubset,"This anonymous record is missing field '%s'."
-tcAnonRecdMultipleFieldsNameSubset,"This anonymous record is missing fields '%s'."
+tcAnonRecdMultipleFieldsNameSubset,"This anonymous record is missing fields %s."
 tcAnonRecdSingleFieldNameSuperset,"This anonymous record has an extra field. Remove field '%s'."
-tcAnonRecdMultipleFieldsNameSuperset,"This anonymous record has extra fields. Remove fields '%s'."
-tcAnonRecdFieldNameDifferent,"This is the wrong anonymous record. It should have the fields %s."
+tcAnonRecdMultipleFieldsNameSuperset,"This anonymous record has extra fields. Remove fields %s."
+tcAnonRecdSingleFieldNameDifferent,"This anonymous record should have field '%s', not '%s'."
+tcAnonRecdMultipleFieldNameDifferent,"This anonymous record should have fields %s, not %s."
```

Comment:
This does not read particularly well in the test expectations - comma is used both as a separator between the fields, as well as the separator between positive and negative part of the sentence.

What about something like this:
```suggestion
tcAnonRecdMultipleFieldNameDifferent,"This anonymous record should have fields %s. The actual fields %s are not matching it."
```

Reply:
Yeah. Then we also need different messages for when it has one vs. multiple fields.

Maybe we could reuse the phrasing from type mismatch?

```fsharp
"This anonymous record should have field '%s' but here has '%s'."
"This anonymous record should have field '%s' but here has fields %s."
"This anonymous record should have fields %s; but here has '%s'."
"This anonymous record should have fields %s; but here has %s."
```

Reply:
Agreed.  It is an error message that people are already familiar with. @T-Gro, any objections?

Reply:
Semicolon is OK for me - just to make it visually clear where list of fields ends

Reply:
Good :)